docs/src/main/sphinx/connector/memsql.rst

```diff
@@ -115,6 +115,164 @@ that catalog name instead of ``singlestore`` in the above examples.
 Type mapping
 ------------
 
+Because Trino and Singlestore each support types that the other does not, this
+connector modifies some types when reading or writing data. Data types may not
+map the same way in both directions between Trino and the data source. Refer to
+the following sections for type mapping in each direction.
+
+Singlestore to Trino read type mapping
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The connector maps Singlestore types to the corresponding Trino types following
+this table:
+
+.. list-table:: Singlestore to Trino type mapping
+  :widths: 30, 20, 50
+  :header-rows: 1
+
+  * - Singlestore database type
+    - Trino type
+    - Notes
+  * - ``BIT``
+    - ``BOOLEAN``
+    -
+  * - ``BOOLEAN``
+    - ``TINYINT``
+    -
+  * - ``TINYINT``
+    - ``TINYINT``
+    -
+  * - ``SMALLINT``
+    - ``SMALLINT``
+    -
+  * - ``INTEGER``
+    - ``INTEGER``
+    -
+  * - ``BIGINT``
+    - ``BIGINT``
+    -
+  * - ``DOUBLE PRECISION``
+    - ``DOUBLE``
+    -
+  * - ``FLOAT``
+    - ``REAL``
+    -
+  * - ``REAL``
+    - ``REAL``
+    -
+  * - ``DECIMAL(p, s)``
+    - ``DECIMAL(p, s)``
+    - See :ref:`Singlestore DECIMAL type handling <singlestore-decimal-handling>`
+  * - ``CHAR(n)``
+    - ``CHAR(n)``
+    -
+  * - ``VARCHAR(n)``
+    - ``VARCHAR(n)``
+    -
+  * - ``TINYTEXT``
+    - ``VARCHAR(255)``
+    -
+  * - ``TEXT``
+    - ``VARCHAR(65535)``
+    -
+  * - ``MEDIUMTEXT``
+    - ``VARCHAR(16777215)``
+    -
+  * - ``LONGTEXT``
+    - ``VARCHAR``
+    -
+  * - ``ENUM(n)``
+    - ``VARCHAR(n)``
+    -
+  * - ``BINARY``, ``VARBINARY``, ``TINYBLOB``, ``BLOB``, ``MEDIUMBLOB``, ``LONGBLOB``
+    - ``VARBINARY``
+    -
+  * - ``DATE``
+    - ``DATE``
+    -
+  * - ``TIME(n)``
+    - ``TIME(n)``
+    -
+  * - ``DATETIME(n)``
+    - ``DATETIME(n)``
+    -
+  * - ``TIMESTAMP(n)``
+    - ``TIMESTAMP(n)``
```

Comment:
No change requested. SingleStore data and time types accepts only no-precision, 0 and 6. 
https://docs.singlestore.com/managed-service/en/reference/sql-reference/data-types/time-and-date.html

Reply:
How is the value converted if it's greater than 6?

Reply:
The connector denies such precisions. 
https://github.com/trinodb/trino/blob/fe16cae089054aa61d1f443c01b4cdc01fd55443/plugin/trino-singlestore/src/main/java/io/trino/plugin/singlestore/SingleStoreClient.java#L433